server/src/main/java/io/unitycatalog/server/persist/ModelRepository.java

```diff
@@ -0,0 +1,363 @@
+                  + registeredModelInfoDAO.getSchemaId());
+        }
+        CatalogInfoDAO catalogInfoDAO =
+            session.get(CatalogInfoDAO.class, schemaInfoDAO.getCatalogId());
+        if (catalogInfoDAO == null) {
+          throw new BaseException(
+              ErrorCode.NOT_FOUND,
+              "Registered model containing catalogId not found: " + schemaInfoDAO.getCatalogId());
+        }
+        registeredModelInfo.setSchemaName(schemaInfoDAO.getName());
+        registeredModelInfo.setCatalogName(catalogInfoDAO.getName());
+        registeredModelInfo.setFullName(
+            getRegisteredModelFullName(
+                catalogInfoDAO.getName(), schemaInfoDAO.getName(), registeredModelInfo.getName()));
+        tx.commit();
+        return registeredModelInfo;
+      } catch (Exception e) {
+        if (tx != null && tx.getStatus().canRollback()) {
+          tx.rollback();
+        }
+        throw e;
+      }
+    }
+  }
+
+  public RegisteredModelInfo getRegisteredModel(String fullName) {
+    LOGGER.info("Getting registered model: " + fullName);
+    RegisteredModelInfo registeredModelInfo = null;
+    try (Session session = SESSION_FACTORY.openSession()) {
+      session.setDefaultReadOnly(true);
+      Transaction tx = session.beginTransaction();
+      try {
+        String[] parts = RepositoryUtils.parseFullName(fullName);
+        String catalogName = parts[0];
+        String schemaName = parts[1];
+        String registeredModelName = parts[2];
+        RegisteredModelInfoDAO registeredModelInfoDAO =
+            findRegisteredModel(session, catalogName, schemaName, registeredModelName);
+        if (registeredModelInfoDAO == null) {
+          throw new BaseException(ErrorCode.NOT_FOUND, "Registered model not found: " + fullName);
+        }
+        registeredModelInfo = registeredModelInfoDAO.toRegisteredModelInfo();
+        registeredModelInfo.setCatalogName(catalogName);
+        registeredModelInfo.setSchemaName(schemaName);
+        registeredModelInfo.setFullName(getRegisteredModelFullName(registeredModelInfo));
+        tx.commit();
+        return registeredModelInfo;
+      } catch (Exception e) {
+        if (tx != null && tx.getStatus().canRollback()) {
+          tx.rollback();
+        }
+        throw e;
+      }
+    }
+  }
+
+  private RegisteredModelInfoDAO findRegisteredModel(
+      Session session, String catalogName, String schemaName, String registeredModelName) {
+    UUID schemaId = getSchemaId(session, catalogName, schemaName);
+    return getRegisteredModelDao(session, schemaId, registeredModelName);
+  }
+
+  public RegisteredModelInfo createRegisteredModel(CreateRegisteredModel createRegisteredModel) {
+    LOGGER.info("Starting createRegisteredModel...");
+    ValidationUtils.validateSqlObjectName(createRegisteredModel.getName());
+    long createTime = System.currentTimeMillis();
+    String modelId = UUID.randomUUID().toString();
+    String storageLocation =
```

Comment:
Will need to change this to instead of just getting the directory, we will need to create the directory, but we will also need to carefully roll that creation back if we fail the create.

Reply:
Will add this in a follow on commit that adds location creation and deals with fixing all of the existing storage root touchpoint code.

Reply:
+1